src/cursor/abstract_cursor.ts

```diff
@@ -653,8 +653,15 @@ function next<T>(cursor: AbstractCursor, blocking: boolean, callback: Callback<T
 
   if (cursorId == null) {
     // All cursors must operate within a session, one must be made implicitly if not explicitly provided
-    if (cursor[kSession] == null && cursor[kTopology].hasSessionSupport()) {
-      cursor[kSession] = cursor[kTopology].startSession({ owner: cursor, explicit: false });
+    if (cursor[kSession] == null) {
+      if (cursor[kTopology].shouldCheckForSessionSupport()) {
+        return cursor[kTopology].selectServer(ReadPreference.primaryPreferred, err => {
```

Comment:
I'm not sure about primaryPreferred here, I think we should make use of `this[kServer]`, and do the self selection thing we did for getMore, if selecting our saved server fails then we can't proceed anyway, correct?

Reply:
Will `this[kServer]` be non-null when `cursorId` is null?

Reply:
primaryPreferred is correct when connected to a replica set, but it's not clear to me how selectServer will behave when the cluster type is sharded or single.  Will a mongos or standalone satisfy this read preference?

Reply:
Ah if the cursorId is null then we haven't contacted the server yet, so `this[kServer]` will be null, good catch, then I suppose we should just respect the readPreference that's in the cursor options. It actually seems like this is not the place we should be starting the session, maybe that's the core of the bug here, executeOperation that's called in FindCursor's _initialize should handle implicit session logic, but we're creating it here in the cursor in an "explicit" style.

Reply:
The cursor's read preference is not the right one to use (it's likely to be `primary`, but any primary or secondary will do.

Reply:
Okay yea I see now, then this is the right approach, ty for chiming in!

Reply:
When `cursorId` is null, `this[kServer]` is null. The problem with using `this[kServer]` is this issue happens *before* the `find` operation executes - we have not selected a server yet.